src/TriggerBinding/SqlTriggerScaleMonitor.cs

```diff
@@ -0,0 +1,184 @@
+            bool isDecreasing = true;
+            for (int index = 0; index < metrics.Length - 1; index++)
+            {
+                isIncreasing = isIncreasing && metrics[index].UnprocessedChangeCount < metrics[index + 1].UnprocessedChangeCount;
+                isDecreasing = isDecreasing && (metrics[index + 1].UnprocessedChangeCount == 0 || metrics[index].UnprocessedChangeCount > metrics[index + 1].UnprocessedChangeCount);
+            }
+
+            if (isIncreasing)
+            {
+                // Scale out only if the expected count of unprocessed changes would exceed the combined limit after 30 seconds.
+                DateTime referenceTime = metrics[metrics.Length - 1].Timestamp - TimeSpan.FromSeconds(30);
+                SqlTriggerMetrics referenceMetric = metrics.First(metric => metric.Timestamp > referenceTime);
+                long expectedUnprocessedChangeCount = (2 * metrics[metrics.Length - 1].UnprocessedChangeCount) - referenceMetric.UnprocessedChangeCount;
+
+                if (expectedUnprocessedChangeCount > workerCount * this._maxChangesPerWorker)
+                {
+                    status.Vote = ScaleVote.ScaleOut;
+                    this._logger.LogInformation($"Requesting scale-out: Found the unprocessed changes for table: '{this._userTable.FullName}' to be continuously increasing" +
+                        " and may exceed the maximum limit set for the workers.");
+                    return status;
+                }
+                else
+                {
+                    this._logger.LogDebugWithThreadId($"Avoiding scale-out: Found the unprocessed changes: {expectedUnprocessedChangeCount} for table: '{this._userTable.FullName}' to be increasing" +
+                        " but they may not exceed the maximum limit set for the workers.");
+                }
+            }
+
+            if (isDecreasing)
+            {
+                // Scale in only if the count of unprocessed changes will not exceed the combined limit post the scale-in operation.
+                if (lastUnprocessedChangeCount <= (workerCount - 1) * this._maxChangesPerWorker)
+                {
+                    status.Vote = ScaleVote.ScaleIn;
+                    this._logger.LogInformation($"Requesting scale-in: Found table: '{this._userTable.FullName}' to be either idle or the unprocessed changes to be continuously decreasing.");
+                    return status;
+                }
+                else
+                {
+                    this._logger.LogDebug($"Avoiding scale-in: Found the unprocessed changes for table: '{this._userTable.FullName}' to be decreasing" +
+                        " but they are high enough to require all existing workers for processing.");
+                }
+            }
+
+            this._logger.LogInformation($"Requesting no-scaling: Found the number of unprocessed changes for table: '{this._userTable.FullName}' to not require scaling.");
```

Comment:
I'm wondering if we should just make these all debug. Is this something users will generally find useful? What do other extensions do for logging related to scaling?

Reply:
I see that Cosmos DB, service bus extensions have logs for scaling at the information level. 
I do think that these logs are helpful for finding out if the scale controller is at fault if the scale decision does not correspond to number of workers allocated for the function.

Reply:
Works for me, easy enough to change later on if we think it's too noisy.